clients/drcachesim/common/trace_entry.h

```diff
@@ -1521,6 +1521,16 @@ typedef struct _pt_data_buf_t pt_data_buf_t;
 
 #endif // defined(BUILD_PT_TRACER) || defined(BUILD_PT_POST_PROCESSOR)
 
+/**
+ * Value used by the system call trace template files (having the type
+ * #OFFLINE_FILE_TYPE_KERNEL_SYSCALL_TRACE_TEMPLATES) in the
+ * #TRACE_MARKER_TYPE_SYSCALL_TRACE_START and
+ * #TRACE_MARKER_TYPE_SYSCALL_TRACE_END markers to denote a trace to
+ * be used for syscalls that have no other trace available in the
+ * template file.
+ */
+constexpr int DEFAULT_SYSCALL_TRACE_TEMPLATE_NUM = 0xffff;
```

Comment:
Should have looked more closely at this value before: this is not a separate enough value to be a sentinel. You can see syscall numbers that are very close to 65535 in [https://github.com/DynamoRIO/drmemory/blob/master/drsyscall/drsyscall_numx.h#L2115](https://www.google.com/url?sa=D&q=https%3A%2F%2Fgithub.com%2FDynamoRIO%2Fdrmemory%2Fblob%2Fmaster%2Fdrsyscall%2Fdrsyscall_numx.h%23L2115) (e.g., 0x10044 = 65604 which is quite close) or [https://github.com/DynamoRIO/dynamorio/blob/master/core/win32/syscallx.h#L140](https://www.google.com/url?sa=D&q=https%3A%2F%2Fgithub.com%2FDynamoRIO%2Fdynamorio%2Fblob%2Fmaster%2Fcore%2Fwin32%2Fsyscallx.h%23L140) 0x20196 etc.  So 65536 is going to look like a legitimate syscall there.  0xFFFFFFFF would be better.